packages/mdc-textfield/_mixins.scss

```diff
@@ -62,3 +62,295 @@
     }
   }
 }
+
+// Private Mixins
+
+// Common
+
+@mixin mdc-text-field-invalid_ {
+  + .mdc-text-field-helper-text--validation-msg {
+    opacity: 1;
+    color: $mdc-text-field-error-on-light;
+  }
+
+  @include mdc-theme-dark(".mdc-text-field", true) {
+    + .mdc-text-field-helper-text--validation-msg {
+      color: $mdc-text-field-error-on-dark;
+    }
+  }
+
+  .mdc-text-field__label {
+    color: $mdc-text-field-error-on-light;
+  }
+
+  .mdc-text-field__input {
+    border-color: $mdc-text-field-error-on-light;
+  }
+
+  .mdc-text-field__bottom-line {
+    background-color: $mdc-text-field-error-on-light;
+  }
+
+  .mdc-text-field__idle-outline {
+    border-color: $mdc-text-field-error-on-light;
+  }
+
+  .mdc-text-field__input:hover ~ .mdc-text-field__idle-outline {
+    border-color: $mdc-text-field-error-on-light;
+  }
+
+  .mdc-text-field__outline .mdc-text-field__outline-path {
+    stroke: $mdc-text-field-error-on-light;
+  }
+}
+
+@mixin mdc-text-field-dense_ {
+  .mdc-text-field--dense {
+    margin-top: 12px;
+    margin-bottom: 4px;
+    font-size: .813rem;
+
+    .mdc-text-field__label--float-above {
+      // NOTE: This is an eyeball'd approximation of what's in the mocks.
+      transform: translateY(calc(-100% - 2px)) scale(.923, .923);
+    }
+  }
+}
+
+@mixin mdc-text-field-disabled_ {
+  .mdc-text-field--disabled {
```

Comment:
You should not include the `.mdc-text-field--disabled` inside the mixin. Instead change mdc-text-field.scss to 

```
.mdc-text-field--disabled {
  @include mdc-text-field-disabled_;
}
```